demisto_sdk/commands/common/content/content.py

```diff
@@ -127,3 +128,143 @@ def content_descriptor(self) -> Optional[ContentDescriptor]:
+                       no_auto_stage=False) -> Set[Path]:
+        prev_ver = prev_ver.replace('origin/', '')
+        content_repo: Repo = self.git()
+
+        # staging all local changes
+        if not no_auto_stage:
+            content_repo.git.add('.')
+        renamed = {item[0] for item in self.renamed_files(prev_ver, committed_only, staged_only)}
+
+        committed = {Path(os.path.join(item.a_path)) for item
+                     in content_repo.remote().refs[prev_ver].commit.diff(
+            content_repo.active_branch).iter_change_type('M')}
+
+        if committed_only:
+            return committed - renamed
+
+        staged = {Path(os.path.join(item.a_path)) for item
+                  in content_repo.head.commit.diff(
+            paths=list((Path().cwd() / 'Packs').glob('*/'))).iter_change_type('M')}
+
+        committed_added = {Path(os.path.join(item.a_path)) for item in content_repo.remote().refs[prev_ver].commit.diff(
+            content_repo.active_branch).iter_change_type('A')}
+
+        staged = staged - committed_added
+
+        if staged_only:
+            return staged - renamed
+
+        return staged.union(committed) - renamed
+
+    def added_files(self, prev_ver='master', committed_only=False, staged_only=False, no_auto_stage=False) -> Set[Path]:
+        prev_ver = prev_ver.replace('origin/', '')
+        content_repo: Repo = self.git()
+
+        # staging all local changes
+        if not no_auto_stage:
+            content_repo.git.add('.')
+
+        committed = {Path(os.path.join(item.a_path)) for item
+                     in content_repo.remote().refs[prev_ver].commit.diff(
+            content_repo.active_branch).iter_change_type('A')}
+
+        if committed_only:
+            return committed
+
+        staged = {Path(os.path.join(item.a_path)) for item in content_repo.head.commit.diff(
+            paths=list((Path().cwd() / 'Packs').glob('*/'))).iter_change_type('A')}
+
+        committed_added_locally_modified = {Path(os.path.join(item.a_path)) for item in content_repo.head.commit.diff(
+            paths=list((Path().cwd() / 'Packs').glob('*/'))).iter_change_type('M')}.intersection(committed)
+
+        staged = staged.union(committed_added_locally_modified)
+
+        if staged_only:
+            return staged
+
+        return staged.union(committed)
+
+    def renamed_files(self, prev_ver='master', committed_only=False, staged_only=False,
+                      no_auto_stage=False) -> Set[Tuple[Path, Path]]:
+        prev_ver = prev_ver.replace('origin/', '')
+        content_repo: Repo = self.git()
+
+        # staging all local changes
+        if not no_auto_stage:
+            content_repo.git.add('.')
+
+        committed = {(Path(item.a_path), Path(item.b_path)) for item
+                     in content_repo.remote().refs[prev_ver].commit.diff(
+            content_repo.active_branch).iter_change_type('R')}
```

Comment:
rename should be considered only when it is 100%

Reply:
what do you mean by 100%? cause the distinction of R vs M is a git standard - I separated the two to be inline with the git terms as well as handle the renamed format (tuple instead of the standard string) as a separate function.